ENIGMAsystem/SHELL/Universal_System/estring.cpp

```diff
@@ -383,4 +393,27 @@ var string_split(const std::string &str, const std::string &delim,
   return res;
 }
 
+string directory_contents(string dname) {
+  if (string_replace_all(dname, " ", "") == "") dname = ".";
+  if (directory_exists(dname)) {
+    string rpath = filename_absolute(dname);
+    #if CURRENT_PLATFORM_ID == OS_WINDOWS
+    if (rpath.back() != '\\') rpath += "\\";
+    #else
+    if (rpath.back() != '/') rpath += "/";
+    #endif
+    string item = file_find_first(rpath + "*", fa_readonly + fa_hidden + fa_sysfile + fa_volumeid + fa_directory + fa_archive);
+    string res = rpath + item;
+    while ((file_exists(rpath + item) || directory_exists(rpath + item)) && item != "") {
```

Comment:
Why do you expect `file_find` to return a path that doesn't exist? By contract, it returns the empty string when iteration is complete.

Reply:
yeah idk man i wasn't thinking i guess